scripts/Muon/GUI/Common/fitting_widgets/basic_fitting/basic_fitting_model.py

```diff
@@ -409,6 +409,15 @@ def do_rebin(self) -> bool:
         """Returns true if rebin is selected within the context."""
         return self.context._do_rebin()
 
+    def x_limits_of_current_dataset(self) -> tuple:
+        """Returns the x data limits of the currently selected dataset."""
+        if self.current_dataset_index is not None:
+            workspace = AnalysisDataService.retrieve(self.current_dataset_name)
```

Comment:
use this instead https://github.com/mantidproject/mantid/blob/master/scripts/Muon/GUI/Common/ADSHandler/ADS_calls.py#L19

then we can keep all of the ADS calls to 1 file